File: go/raindrops/raindrops.go
package raindrops

import (
	"fmt"
)

const testVersion = 3

var factors = []int{3, 5, 7}
var conversions = map[int]string{
	3: "Pling",
	5: "Plang",
	7: "Plong",
}

// Convert takes and input number, and returns a string of the number converted to raindrop-speak
func Convert(number int) (output string) {
	for _, factor := range factors {
		if number%factor == 0 {
			output += conversions[factor]
		}
	}

	if output == "" {
		output = fmt.Sprint(number)
	}
	return
}

File: lua/raindrops/raindrops.lua
local factors = {
  { factor = 3, sound = "Pling" },
  { factor = 5, sound = "Plang" },
  { factor = 7, sound = "Plong" },
}

return function (number)
  local text = ''

  for _, f in ipairs(factors) do
    if (number % f.factor) == 0 then text = text .. f.sound end
  end

  return #text > 0 and text or tostring(number)
end
